src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -1172,5 +1172,10 @@ ShenandoahGenerationalHeap::TransferResult ShenandoahFullGC::phase5_epilog() {
     result = ShenandoahGenerationalFullGC::balance_generations_after_rebuilding_free_set();
     ShenandoahGenerationalFullGC::rebuild_remembered_set(heap);
   }
+
+  // passive mode with ShenandoahCardBarrier turned on, clean the write table without swapping the tables
+  if (ShenandoahCardBarrier && !heap->mode()->is_generational()) {
```

Comment:
In general, the passive mode will only initiate a _degenerated_ cycle. It will run a full GC only if the degenerated cycle cannot evacuate or it doesn't make "good progress". Could we move this behavior into `shDegeneratedGC` or somewhere in `shPassiveHeuristic`? Cleaning the card table on every GC would be more aligned with the concurrent mode behavior.